code/ui/generalhud/inspectmenu/InspectMenu.cs

```diff
@@ -174,31 +179,31 @@ public void SetConfirmationData(ConfirmationData confirmationData)
 
                     _timeSinceDeath?.Delete(true);
 
-                    _timeSinceDeath = new InspectItem(this);
+                    _timeSinceDeath = new InspectItem(this._itemWrapper);
                     _timeSinceDeath.ImageWrapper.Image.SetTexture($"/ui/inspectmenu/time.png");
                     _timeSinceDeath.InspectItemLabel.Text = "";
 
                     _headshot?.Delete(true);
 
                     if (confirmationData.Headshot)
                     {
-                        _headshot = new InspectItem(this);
+                        _headshot = new InspectItem(this._itemWrapper);
                         _headshot.ImageWrapper.Image.SetTexture($"/ui/inspectmenu/headshot.png");
-                        _headshot.InspectItemLabel.Text = "By a headshot";
+                        _headshot.InspectItemLabel.Text = "Headshot";
                     }
 
                     _distance?.Delete(true);
                     _suicide?.Delete(true);
 
                     if (!confirmationData.Suicide)
                     {
-                        _distance = new InspectItem(this);
+                        _distance = new InspectItem(this._itemWrapper);
                         _distance.ImageWrapper.Image.SetTexture($"/ui/inspectmenu/distance.png");
-                        _distance.InspectItemLabel.Text = $"From {confirmationData.Distance:n0}m away";
+                        _distance.InspectItemLabel.Text = $"{confirmationData.Distance:n0}m distance";
```

Comment:
"Xm distance" does not sufficiently convey the idea that the number displayed is the distance from which they were killed. I don't think there is any need to change it.

Reply:
This is WIP so unfinishef, I have to add a better design for this to show more information on a smaller space as the current design here is 💩.